hardware/arduino/sam/system/libsam/source/twi.c

```diff
@@ -179,6 +166,27 @@ void TWI_ConfigureSlave(Twi *pTwi, uint8_t slaveAddress)
 }
 
 /**
+ * \brief Disables the TWI.
+ * \param pTwi  Pointer to an Twi instance.
+ */
+void TWI_Disable(Twi *pTwi)
+{
+    assert( pTwi ) ;
+
+    uint32_t i;
+
+    /* TWI software reset */
+    pTwi->TWI_CR = TWI_CR_SWRST;
+    pTwi->TWI_RHR;
+
+    /* Wait at least 10 ms */
+    for (i=0; i < 1000000; i++);
```

Comment:
@cmaglie I would agree. When I try the following sketch, ``millis`` does not increase at all.

```arduino
#include <Wire.h>

void setup() {
  unsigned long start;
  unsigned long afterBegin;
  unsigned long afterEnd;
  
  Serial.begin(9600);

  start = millis();
  Wire.begin(42);
  afterBegin = millis();
  Wire.end();
  afterEnd = millis();

  Serial.println(start);
  Serial.println(afterBegin);
  Serial.println(afterEnd);
}

void loop() {
  unsigned long l = millis();

  Serial.println(l);
  delay(1000);              // wait for a second
}
```

Output was:
```
2
2
2
2
1002
2002
...
```

There are two "delay" loops which are run, this one and one in ```TWI_ConfigureSlave```.